### PR TITLE
bsim: Add EDTT hci test app

### PR DESCRIPTION
**Pull request by mkasenberg**
adopted from Zephyr.

### CODE REVIEW

File: babblesim/apps/edtt_test/transport/edtt/pkg.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+pkg.name: babblesim/apps/edtt_test/transport/edtt
+pkg.description: XXX
```
Comment (sjanc) [inline babblesim/apps/edtt_test/transport/edtt/pkg.yml:21]: give some meaningful description or leave it empty

**Review by sjanc — APPROVED** [review]

File: babblesim/apps/edtt_test/hci_test_app/pkg.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+pkg.keywords:
+
+pkg.deps:
+    - "@apache-mynewt-core/sys/console/stub"
+    - "@apache-mynewt-core/sys/log/stub"
+    - "@apache-mynewt-core/sys/stats/full"
+    - "@apache-mynewt-core/kernel/os"
+    - nimble/controller
+    - nimble/transport
+    - babblesim/apps/edtt_test/transport/edtt
```
Comment (andrzej-kaczmarek) [inline babblesim/apps/edtt_test/hci_test_app/pkg.yml:33]: app pkg should depend on nimble/transport and transport/edtt only, then transport should depend on nimble/controller

File: babblesim/apps/edtt_test/hci_test_app/syscfg.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+syscfg.vals:
+    BLE_HOST: 0
+    BLE_HCI_TRANSPORT: custom
+    MCU_UART_POLLER_PRIO: 3
```
Comment (andrzej-kaczmarek) [inline babblesim/apps/edtt_test/hci_test_app/syscfg.yml:23]: I think we could just change uart poller prio in bsim bsp to smth like 9, I assume it should have lowest priority of all tasks. Then edtt poller can be set to 2 (as it is now) and there's no need to override uart.

File: babblesim/apps/edtt_test/transport/edtt/pkg.yml [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+pkg.description: Provides communication with EDTT bridge over named pipes
+pkg.author: "Apache Mynewt <dev@mynewt.apache.org>"
+pkg.homepage: "http://mynewt.apache.org/"
+pkg.deps:
+    - "@apache-mynewt-core/sys/console/stub"
+    - "@apache-mynewt-core/sys/log/stub"
+    - "@apache-mynewt-core/sys/stats/full"
+    - "@apache-mynewt-core/kernel/os"
+    - nimble/controller
+    - nimble/transport
```
Comment (andrzej-kaczmarek) [inline babblesim/apps/edtt_test/transport/edtt/pkg.yml:30]: should only depend on nimble/controller

File: babblesim/apps/edtt_test/hci_test_app/pkg.yml [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+pkg.req_apis:
+    - ble_transport
```
Comment (andrzej-kaczmarek) [inline babblesim/apps/edtt_test/hci_test_app/pkg.yml:36]: not needed, this is already required by nimble/transport

File: babblesim/apps/edtt_test/transport/edtt/pkg.yml [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+pkg.apis:
+    - ble_transport
+
+pkg.init:
+    ble_hci_edtt_init: 'MYNEWT_VAL(BLE_TRANS_EDTT_SYSINIT_STAGE)'
```
Comment (andrzej-kaczmarek) [inline babblesim/apps/edtt_test/transport/edtt/pkg.yml:36]: just add static priority

File: babblesim/apps/edtt_test/transport/edtt/syscfg.yml [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+        description: >
+            Path to the HCI log file. Skip file extension
+            because device id and .log will be appended.
+        value: "hci_logs"
+    MCU_EDTT_POLLER_PRIO:
+        description: 'Priority of native EDTT poller task.'
+        type: task_priority
+        value: 2
+    BLE_HCI_EDTT_PORT:
+        description: 'The port to use for the HCI edtt interface'
```
Comment (andrzej-kaczmarek) [inline babblesim/apps/edtt_test/transport/edtt/syscfg.yml:31]: EDTT_HCI_PORT

File: babblesim/apps/edtt_test/transport/edtt/syscfg.yml [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+# under the License.
+#
+
+syscfg.defs:
+    EDTT_HCI_LOG_FILE:
+        description: >
+            Path to the HCI log file. Skip file extension
+            because device id and .log will be appended.
+        value: "hci_logs"
+    MCU_EDTT_POLLER_PRIO:
```
Comment (andrzej-kaczmarek) [inline babblesim/apps/edtt_test/transport/edtt/syscfg.yml:26]: EDTT_POLLER_PRIO

File: babblesim/apps/edtt_test/transport/edtt/syscfg.yml [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+            Sysinit stage for the UART BLE transport.
+        value: 500
+
+syscfg.vals.BLE_EXT_ADV:
+    BLE_HCI_EVT_BUF_SIZE: 257
+
+syscfg.vals:
+    BLE_ACL_BUF_COUNT: 32
+    BLE_ACL_BUF_SIZE: 255
+    BLE_HCI_EVT_BUF_SIZE: 70
```
Comment (andrzej-kaczmarek) [inline babblesim/apps/edtt_test/transport/edtt/syscfg.yml:55]: just set to 257 always, we have plenty of ram on native target

File: babblesim/targets/edtthci/pkg.yml [hunk 10]
```diff
@@ -0,0 +1,5 @@
+pkg.name: babblesim/targets/edtthci
```
Comment (andrzej-kaczmarek) [inline babblesim/targets/edtthci/pkg.yml:1]: missing license

File: babblesim/targets/edtthci/syscfg.yml [hunk 11]
```diff
@@ -0,0 +1,10 @@
+syscfg.vals:
```
Comment (andrzej-kaczmarek) [inline babblesim/targets/edtthci/syscfg.yml:1]: missing license

File: babblesim/targets/edtthci/syscfg.yml [hunk 12]
```diff
@@ -0,0 +1,10 @@
...
+  # EDTT requires 0x123456789ABC address for the first device
+  # and 0x456789ABCDEF for the second one
+  BLE_LL_PUBLIC_DEV_ADDR: 0x123456789ABC  # d=1
+  # BLE_LL_PUBLIC_DEV_ADDR: 0x456789ABCDEF # d=2
+
+  # Default task settings
+  OS_MAIN_STACK_SIZE: 8000
```
Comment (andrzej-kaczmarek) [inline babblesim/targets/edtthci/syscfg.yml:8]: not needed

File: babblesim/targets/edtthci/syscfg.yml [hunk 13]
```diff
@@ -0,0 +1,10 @@
...
+  EDTT_HCI_LOG_FILE: ("hci_logs")
+  EDTT_HCI_LOGS: 1
```
Comment (andrzej-kaczmarek) [inline babblesim/targets/edtthci/syscfg.yml:10]: this does not seem to be defined anywhere, I guess it should be used instead of -DEDTT_HCI_LOGS ?

**Review by andrzej-kaczmarek — CHANGES_REQUESTED** [review]

**Comment by andrzej-kaczmarek** [comment]
also I'd simplify directory structure:
babblesim/edtt/hci_test_app
babblesim/edtt/hci_transport

**Review by andrzej-kaczmarek — APPROVED** [review]

**Pull request merged**